### PR TITLE
chore(std/http): server module name migration

### PR DESCRIPTION
**Pull request by cmorten**
Relates to https://github.com/denoland/deno_std/discussions/1034 and https://github.com/denoland/deno_std/pull/1128.

For expedience I have made the simplest change of migrating `server.ts` -> `server_legacy.ts`. It would be good to use the new std/http APIs for these tests in future - happy to try rework these, but will take a little longer to do. If we do go with this, we can raise an issue to cover this off at a slower cadence.

### CODE REVIEW

**Comment by cmorten** [comment]
I _believe_ this won’t pass CI until the next version of std is released.

File: cli/tests/testdata/module_graph/file_tests-checkwithconfig.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { ServerRequest } from "https://deno.land/std/http/server.ts";
+import { ServerRequest } from "https://deno.land/std/http/server_legacy.ts";
```
Comment (bartlomieju) [inline cli/tests/testdata/module_graph/file_tests-checkwithconfig.ts:1]: Yikes, this file should use `test_util/std/` instead of remote server.

**Review by bartlomieju — COMMENTED** [review]
@cmorten `test_util/std/` is a git submodule, so we don't have to wait for std release to update it. Let me know if you want to try and do that or should I

File: cli/tests/testdata/module_graph/file_tests-checkwithconfig.ts (hunk 1 above)
Comment (cmorten) [reply cli/tests/testdata/module_graph/file_tests-checkwithconfig.ts:1]: That’ll be why then 😄 was confused by the inconsistency 😅

**Comment by cmorten** [comment]
Hmm _I believe_ I've updated the submodule ok @bartlomieju, but looks like there are some errors thrown in the format tests - the result of some changes @kitsonk made prior to the "native" https server work in std for TypeScript 4.4 support https://github.com/denoland/deno_std/pull/1171

Not sure how best to proceed? Happy to raise a PR to try resolve the type issues in std as seen by this repo (but not std's CI)?

**Comment by bartlomieju** [comment]
> Hmm _I believe_ I've updated the submodule ok @bartlomieju (please shout if not!), but looks like there are some errors thrown in the format tests - the result of some changes @kitsonk made prior to the "native" HTTP server work in std for TypeScript 4.4 support [denoland/deno_std#1171](https://github.com/denoland/deno_std/pull/1171).
> 
> Not sure how best to proceed? Happy to raise a PR to try resolve the type issues in std as seen by this repo (but not std's CI)?

Yes, format and lint script use latest deno binary and the TypeScript update is not yet available there. I guess we'll have to wait for 1.14 release before we can update `deno_std` submodule.

**Comment by cmorten** [comment]
> Yes, format and lint script use latest deno binary and the TypeScript update is not yet available there. I guess we'll have to wait for 1.14 release before we can update `deno_std` submodule.

Curiously in CI, the lint steps were fixed to `v1.11.3` which is 2 minor versions behind latest? (last updated in https://github.com/denoland/deno/pull/11369). I've updated for the purpose of this PR - let me know if any reason why couldn't bump to the latest!

Seems this just uncovers the old test issue which thought would be solved by the std submodule update, so will have a debug locally to see what's going on...

**Comment by cmorten** [comment]
Ok think it's sorted - I had misunderstood the `file_tests-checkwithconfig.ts` test's use of

```ts
import { ServerRequest } from "https://deno.land/std/http/server.ts";
```

This _isn't_ going to the std lib, but instead resolves to a local `https_deno.land-std-http-server.ts` mock via importmap. See https://github.com/denoland/deno/pull/8167 and https://github.com/denoland/deno/issues/8163 for details. We _could_ consider renaming the import so to not to cause the same confusion in future, but not sure if that detracts from the test - I suspect it is fine, and actually looking at the blame + test setup quickly pointed to the purpose.

**Comment by cmorten** [comment]
@bartlomieju all issues resolved now 🤞🏻

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks for looking into this Craig!

**Pull request merged**